modules/ballerina-core/src/main/java/org/ballerinalang/model/types/TypeLattice.java

```diff
@@ -51,7 +55,10 @@ public static TypeLattice getExplicitCastLattice() {
     public static TypeLattice getImplicitCastLattice() {
         return implicitCastLattice;
     }
-
+    
+    public static TypeLattice getTransformLattice() {
+        return conversionLattice;
+    }
 
     public static void loadImplicitCastLattice(SymbolScope scope) {
```

Comment:
+1.
Only concern is the json scenario, where with implicit casting `json j = 5;`  is possible.  But if we remove it we have to enforce `json j = (json) 5;`